File: samples/19_three-layer-kmp-domain-presentation/androidApp/src/main/kotlin/com/example/kmpsamples/threelayerkmpdomainpresentation/android/TaskRepository.kt
package com.example.kmpsamples.threelayerkmpdomainpresentation.android

import com.example.kmpsamples.threelayerkmpdomainpresentation.shareddomain.Task
import com.example.kmpsamples.threelayerkmpdomainpresentation.shareddomain.TaskRepository as SharedTaskRepository

class TaskRepository(
    private val dataSource: TaskDataSource
) : SharedTaskRepository {
    override fun loadTask(platform: String): Task = Task(
        title = "Android repository task",
        details = dataSource.loadTask(platform)
    )
}

File: samples/three-layer-kmp-domain-presentation/iosApp/Sources/TaskRepository.swift
import SharedPresentation

final class IosTaskRepository: TaskRepository {
    private let dataSource: TaskDataSource

    init(dataSource: TaskDataSource) {
        self.dataSource = dataSource
    }

    func loadTask(platform: String) -> Task {
        Task(
            title: "iOS repository task",
            details: dataSource.loadTask(platform: platform)
        )
    }
}
